hack/mock-unittest-data/yarn.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+import itertools
+import json
+import pprint
+import re
+import subprocess
+import sys
+import tempfile
+import textwrap
+from pathlib import Path
+from typing import Any, Iterator, Optional
+
+
+def print_banner(content: str) -> None:
+    print("-" * 80)
+    print("\n".join(textwrap.wrap(content, 80)))
+    print("-" * 80)
+
+
+def clone_repo(tmpdir: Path) -> Path:
+    repo_dir = tmpdir / "cachi2-yarn-berry"
+    subprocess.run(
+        [
+            "git",
+            "clone",
+            "https://github.com/cachito-testing/cachi2-yarn-berry",
+            "--depth=1",
+            "--single-branch",
+            "--branch=zero-installs",
+            repo_dir,
+        ],
+        check=True,
+    )
+    return repo_dir
+
+
+def run_yarninfo(repo_dir: Path) -> str:
+    proc = subprocess.run(
+        ["yarn", "info", "--all", "--recursive", "--json", "--cache"],
+        cwd=repo_dir,
+        stdout=subprocess.PIPE,
+        text=True,
+    )
+    try:
+        proc.check_returncode()
+    except subprocess.CalledProcessError:
+        # yarn prints errors to stdout
+        print(proc.stdout, file=sys.stderr)
+        raise
```

Comment:
I wasn't really planning to bother with nice errors. This is the only extra handling because yarn, at the moment, has nothing in stderr.

For the git subprocess, I'm letting it print everything directly to stdout/stderr without capturing, for yarn I need the stdout so I at least re-print it on failure